src/js-form-validator.ts

```diff
@@ -313,10 +313,8 @@ export class FormValidator {
       case 'phone':
         if (!allowEmpty && !value.length) return -1;
         let vals = value.split('x');
-        for (let i = 0, l = vals.length; i < l; i++)
-          vals[i] = vals[i].replace(/\D/g, '');
         value = vals.join('x').replace(/^1/, ''); // remove leading 1 if added, only for US
-        status = /^[0-9x]{7,16}$/.test(value) ? 1 : 0;
+        status = /^[0-9\-\+\s\(\)]{7,16}$/.test(value) ? 1 : 0;
```

Comment:
@someyoungideas ah interesting.

What if we came at this in a more usable way. Ultimately, we don't care about anything else but the numbers. We also don't care about extensions, so we can just remove that support to make the validation easier.

So, knowing that, should we develop a small web component which handles input for a "tel" type HTML input, validates only the numbers, and then formats it to whatever automatically on blur?

So, if they end up putting in 28s-840-8006, it will auto strip out the _s_, and then validate and find out it's still invalid. If they put in 128s-840-8006, it would strip out the s and then find it's actually valid.

Thoughts? cc @samiam236